lib/ui/pages/account-linking/web_auth.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:flutter_web_auth/flutter_web_auth.dart';
+import 'package:pispapp/controllers/ephemeral/account-linking/web_auth_controller.dart';
+import 'package:pispapp/controllers/flow/account_linking_flow_controller.dart';
+import 'package:pispapp/ui/theme/light_theme.dart';
+import 'package:pispapp/ui/widgets/bottom_button.dart';
+import 'package:pispapp/ui/widgets/title_text.dart';
+
+class WebAuth extends StatelessWidget {
+  WebAuth(this._accountLinkingFlowController);
+
+  final AccountLinkingFlowController _accountLinkingFlowController;
+  final WebAuthController _authController = WebAuthController();
+
+  Widget _buildActionSection() {
+    return GetBuilder<AccountLinkingFlowController>(
+      init: _accountLinkingFlowController,
+      global: false,
+      builder: (controller) {
+        if (!controller.isAwaitingUpdate) {
+          // The user is only allowed to click the button once.
+          // Afterward, the state of the payment flow controller will be updated
+          // to be awaiting for response. Once the response is received, the
+          // flow controller is expected to bring used to another page.
+          return BottomButton(
+            const TitleText(
+              'Log in with financial service provider',
+              color: Colors.white,
+              fontSize: 15,
+            ),
+            onTap: () async {
+              final String result = await FlutterWebAuth.authenticate(
+                  url: controller.consent.authUri,
+                  callbackUrlScheme: WebAuthController.customScheme);
+              final String authToken =
+                  _authController.extractAuthTokenFromResult(result);
+              if (authToken != null) {
```

Comment:
maybe can do something if the authToken is null? Or perhaps leave a TODO there

Reply:
I will leave a TODO.
If authToken is null, then DFSP didnt implement it correctly OR the user was not authenticated.